lib/bitcoin/SpendingMonitor.ts

```diff
@@ -0,0 +1,113 @@
+
+    if (bitcoinFeeSpendingCutoffPeriodInBlocks < 1 || bitcoinFeeSpendingCutoffPeriodInBlocks > SpendingMonitor.estimatedBlockWritesInOneMonth) {
+      // tslint:disable-next-line: max-line-length
+      throw new Error(`Bitcoin spending cutoff period: ${bitcoinFeeSpendingCutoffPeriodInBlocks} must be between 1-${SpendingMonitor.estimatedBlockWritesInOneMonth}`);
+    }
+
+    if (bitcoinFeeSpendingCutoffInSatoshis <= 0) {
+      throw new Error('Bitcoin spending cutoff amount must be > 0');
+    }
+
+    this.anchorStringsWritten = new Set<string>();
+  }
+
+  /**
+   * Add the transaction data to track the spending.
+   * @param anchorString The string to be written.
+   */
+  public addTransactionDataBeingWritten (anchorString: string): void {
+    this.anchorStringsWritten.add(anchorString);
+  }
+
+  /**
+   * Calculates whether the specified fee will result in this node going over the spending limit.
+   * @param currentFeeInSatoshis The fee to be added for the next transaction.
+   * @param startingBlockHeight The block height to start the check for the cutoff period.
+   */
+  public async isCurrentFeeOverSpendingLimit (currentFeeInSatoshis: number, lastProcessedBlockHeight: number): Promise<boolean> {
+
+    // Special case for when the checking period is 1. Even though the algorithm later will
+    // will work for this case, but we will avoid making a DB call.
+    if (this.bitcoinFeeSpendingCutoffPeriodInBlocks === 1) {
+      return (currentFeeInSatoshis > this.bitcoinFeeSpendingCutoffInSatoshis);
+    }
+
+    // In order to calculate whether we are over the spending limit or not, our algorithm is:
+    // <code>
+    //   feesFromCutoffPeriod = transactionStore.get_fees_for_cutoff_period
+    //   totalFees = currentFee + feesFromCutoffPeriod
+    //   if (totalFees > spendingCutoff) { return true }
+    // </code>
+    //
+    // Now remember that the currentFee input is for the next block (which is not written yet), so
+    // the next block is included in the cutoff period. Also, the input lastProcessedBlockHeight is
+    // also included in the cutoff period. So when we go back to the transaction store, we subtract
+    // the these 2 blocks from the cutoff period. For example:
+    //  - if the cutoff period is 2, then we want transactions from the last-processed-block and the next one.
+    //  - if the cutoff period is 3, then we want transactions from the last-process-block - 1, last-processed-block, and the next one
+    const startingBlockHeight = lastProcessedBlockHeight - this.bitcoinFeeSpendingCutoffPeriodInBlocks - 2;
+
+    // Now get the transactions from the store which are included in the above starting block and higher.
+    const startingBlockFirstTxnNumber = TransactionNumber.construct(startingBlockHeight, 0);
+
+    const allTxnsSinceStartingBlock =
+      await this.transactionStore.getTransactionsLaterThan(startingBlockFirstTxnNumber - 1, SpendingMonitor.maxResultsFromTransactionStore);

```

Comment:
Consider changing the `getTransactionsLaterThan()` interface method so that it can take only the first parameter. This way `maxResultsFromTransactionStore` won't be needed and the spending calculation will not yield false positive allow-writes.